reference/architecture/authz_authn/authorization/index.md

```diff
@@ -0,0 +1,134 @@
+roles. Fiat instances coordinate around locks (in Redis) to ensure that only one instance synchronizes roles at a time.
+
+## Implementation Details
+
+### Roles and Permissions
+
+Fiat uses the following model for user permissions:
+
+- a user ID (= a real one or [`__unrestricted_user__`](#unrestricted-user)  )
+- Accounts permission = list of { name + cloudProvider + Permissions)
+- Apps permissions = list of { name + permissions }
+- Service accounts = list of service accounts the user belongs to
+- Roles: list of roles the user has
+- build services: list of build services the user has access to
+
+### Sync
+
+Every 30 seconds, Fiat checks if it needs to sync roles. Every 10 minutes (by default), it will sync user ↔ roles. It may mean querying the provider for all the roles of the all the users that Fiat knows about (= are cached in Redis).
+
+#### Unrestricted user
+
+- At any point, the unrestricted virtual user should have `UserPermission` in the permission repository (Redis for now) for all accounts, apps, service accounts, build services that have not been restricted (no permission specified).
+- The unrestricted user’s permissions is updated on every sync to account for permission changes as well as new for new apps, accounts, etc.
+- When returning a user’s permission, it is merged with the unrestricted user. By having the account, app, service account, build service in the `UserPermission` of the user, it is known and the default access for unrestricted app/.. should apply.
+
+#### Note:
+
+During the sync while reading apps permissions from Front50 (in the app definition), Fiat checks if the app has roles defined for `EXECUTE`. If not, Fiat copies the list of roles defined on the app for `fiat.executeFallback` (which can be `READ` or `WRITE`) to the `EXECUTE` permission list. This is done to ensure that at least some roles can execute a pipeline as that role has been introduced recently.
+
+### Verifying access from a service
+
+A service checks if user `userId` has permission `P`  on resource `R` of type `T` (apps, account, build service). The following steps take place in the service calling Fiat. The response is detailed thereafter.
+
+If Fiat is not enabled → Yes
+If Fiat is enabled → query Fiat with `userId`:
+
+- Check the local cache first (that expires after `services.fiat.cache.expiresAfterWriteSeconds` defaults to 20)
+- If the request fails, retry will backoff, if it keeps failing:
+    - if `services.fiat.legacyFallback = true`:
+        - If `T == account`: if the account has at least one `WRITE` permission, Yes (TO BE CONFIRMED), otherwise No
+        - if `T == app`: Yes (note: via `allowAccessToUnknownApplications`)
+        - if `T == buildService`: Yes
+    - else reject
+- if the request succeeds, permissions are returned by Fiat:
+    - See +Deep Dive… Fiat: Permissions-returned-by-Fiat  how Fiat computes the response.
```

Comment:
See above comment - I'm removing the line, but yeah this entire page could use cleanup.  I'm a believer in get it out there where it can be used and improve as we go.